include/uuid/UUIDv8.h

```diff
@@ -0,0 +1,320 @@
+#define _UUIDV8_H_
+
+#include <uuid/uuid.h>
+
+#include <algorithm>
+#include <bitset>
+#include <chrono>
+#include <cstddef>
+#include <ios>
+#include <iostream>
+#include <memory>
+#include <random>
+#include <sstream>
+#include <string>
+
+#include "random_gen.h"
+
+namespace uprotocol::uuid::factory {
+
+/**
+ *
+ * UUIv8Factory generates UUID in v8 format and contain various utility methods
+for UUIDv8
+ *
+ * UUIDv8 offers variable-size timestamp, clock sequence, and node values which
+allow
+ * for a highly customizable UUID that fits a given application needs.
+ *
+ * More information on
+ * https://www.ietf.org/archive/id/draft-peabody-dispatch-new-uuid-format-01.html#name-uuidv8-layout-and-bit-order
+ *
+ *
+ * // UUID V8 Format
+// 0                   1                   2                   3
+// 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+//+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//| custom_a |
+//+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//| custom_a | ver | custom_b |
+//+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//|var| custom_c |
+//+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//| custom_c |
+//+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ *
+*/
+class UUIDv8 {
+public:
+    /**
+     * Constructor to build UUIDv8 object
+    */
+    UUIDv8();
+
+    /**
+     * @brief method will create the msb and lsb parts of UUIDv8 format
+     * @param now time at which UUIDv8 number is requested for checking previous UUIDv8
+     * @param uuid_prev  if exist previous UUIDv8 number for tracking pat history
+    */
+    void generate(uint64_t now, UUIDv8 *uuidPrev = nullptr);
+
+    /**
+     * @brief This method copies current UUIDv8 object to destination UUIDv8 object that's
+    *   sent as parameter
+     * @param[out] dest UUIDv8 object is the target object
+    */
+    inline void copy(UUIDv8& dest) const {
+        dest.msb_ = this->msb_;
+        dest.lsb_ = this->lsb_;
+    }
+
+    /**
+     ** @brief Utility function copies org UUIDv8 object to destination UUIDv8 object that's
+    *   sent as parameter
+    *   @param[in] org UUIDv8 object is the source object
+     * @param[out] dest UUIDv8 object is the target object
+    */
+    static inline void copy(UUIDv8& dest, UUIDv8 const& org) {
+        dest.msb_ = org.msb_;
+        dest.lsb_ = org.lsb_;
+    }
+
+    /**
+     * @brief converts the cuurent UUIDv8 id to string format
+     * @return UUIDv8 string
+     */
+    std::string toString() const;
+
+    /**
+     * @brief uuidV8ToString utility function to convert given UUIDv8 number array to string format
+     * @param  uuid UUIDV8 number in array
+     * @return UUIDv8 string
+    */
+    static inline std::string uuidV8ToString(const uint8_t* uuid) {
+        static const char DIGITS[] = "0123456789abcdef";
+        std::string str{};
+        for (int i = 0; i < 16; i++) {
```

Comment:
Can we have a constant instead of using the magic number 16

Reply:
Fixed